lua/sql/parser.lua

```diff
@@ -334,4 +332,54 @@ M.drop = function(tbl)
   return string.format("drop table %s", tbl)
 end
 
+---Preporcess data insert to sql db.
+---for now it's mainly used to for parsing lua tables and boolean values.
+---It throws when a schema key is required and doesn't exists.
+---@param rows table inserted row.
+---@param schema table tbl schema with extra info
+---@return table pre processed rows
+M.pre_insert = function(rows, schema)
+  rows = u.is_nested(rows) and rows or { rows }
+
+  for _, row in ipairs(rows) do
+    for k, _ in pairs(schema.req) do
+      if not row[k] then
+        error("sql.nvim: (insert) missing a required key: " .. k)
+      end
+    end
+
+    for k, v in pairs(row) do
+      if schema.types[k] == "luatable" or schema.types[k] == "json" then
+        row[k] = json.encode(v)
```

Comment:
does this mean using your library will internally 100% use json?!?!

If so i am actually considering doing it manually. This is a unnecessary performance hit

Reply:
nope, if a schema definition doesn't have luatable or "json" as key value it will act as if json doesn't exists.

maybe I should have `require"json".encode(v)` to be extra sure

Reply:
I dont get how any of this works

Reply:
i see now. nvm